### PR TITLE
Adding a runtimeconfig.dev.json containing the additionalProbePaths when doing dotnet build.

### PR DESCRIPTION
**Pull request by livarcocc**
If I finish the project.json runtimeoptions work soon enough, I will put it here as well.

cc @anurse @pakrym @brthor @schellap @gkhanna79 @piotrpMSFT

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2130)

### CODE REVIEW

**Comment by livarcocc** [comment]
Fixes https://github.com/dotnet/cli/issues/2090.

**Comment by livarcocc** [comment]
@dotnet-bot Test OSX x64 Release Build

File: test/dotnet-build.Tests/BuildPortableTests.cs [hunk 1]
```diff
@@ -6,8 +6,9 @@ namespace Microsoft.DotNet.Tools.Builder.Tests
 {
     public class BuildPortableTests : TestBase
     {
-        [Fact]
-        public void BuildingAPortableProjectProducesDepsFile()
+        private readonly DirectoryInfo _netstandardappOutput;
+        
+        public BuildPortableTests()
```
Comment (piotrpMSFT) [inline test/dotnet-build.Tests/BuildPortableTests.cs]: looks like we dropped a [Fact]

File: test/dotnet-build.Tests/BuildPortableTests.cs [hunk 2]
```diff
@@ -20,17 +21,43 @@ public void BuildingAPortableProjectProducesDepsFile()
...
+        [Fact]
+        public void BuildingAPortableProjectProducesARuntimeConfigJsonFile()
+        {
+            _netstandardappOutput.Should().Exist().And.HaveFile("PortableApp.runtimeconfig.json");
+        }
+        
+        [Fact]
+        public void BuildingAPortableProjectProducesARuntimeConfigDevJsonFile()
+        {
+            _netstandardappOutput.Should().Exist().And.HaveFile("PortableApp.runtimeconfig.dev.json");
```
Comment (piotrpMSFT) [inline test/dotnet-build.Tests/BuildPortableTests.cs]: Do we need this for portable apps? Or just standalone?

**Comment by piotrpMSFT** [comment]
@schellap can you verify? 

I believe the runtimeconfig.dev.json file should only be produced when building a standalone application, and not for portable applications.

File: test/dotnet-build.Tests/BuildPortableTests.cs (hunk 2 above)
Comment (davidfowl) [inline test/dotnet-build.Tests/BuildPortableTests.cs]: Just stand alone apps.

Comment (schellap) [inline test/dotnet-build.Tests/BuildPortableTests.cs]: Where does portable's non core fx dependencies come from without the additionalProbingPath?

Even portable app should work with `dotnet portable.dll`

Comment (gkhanna79) [inline test/dotnet-build.Tests/BuildPortableTests.cs]: Correct - the runtimeconfig.dev.json is required for both portable and standalone apps to run after simply doing a "dotnet build".

**Comment by gkhanna79** [comment]
@dotnet-bot test this please

File: test/dotnet-build.Tests/BuildPortableTests.cs [hunk 3]
```diff
@@ -20,17 +21,43 @@ public void BuildingAPortableProjectProducesDepsFile()
...
+        }
+        
+        [Fact]
+        public void BuildingAPortableProjectProducesARuntimeConfigJsonFile()
+        {
+            _netstandardappOutput.Should().Exist().And.HaveFile("PortableApp.runtimeconfig.json");
+        }
+        
+        [Fact]
+        public void BuildingAPortableProjectProducesARuntimeConfigDevJsonFile()
```
Comment (gkhanna79) [inline test/dotnet-build.Tests/BuildPortableTests.cs]: Is there a similar test for standalone apps to validate that this file is being produced for both the scenarios?

File: test/dotnet-publish.Tests/PublishPortableTests.cs [hunk 4]
```diff
@@ -74,5 +80,11 @@ public void PortableAppWithIntentionalDowngradePublishesDowngradedManagedCode()
                 "System.Linq.dll"
             });
         }
+        
+        [Fact]
+        public void PortableAppWithRuntimeTargetsDoesNotHaveRuntimeConfigDevJsonFile()
```
Comment (gkhanna79) [inline test/dotnet-publish.Tests/PublishPortableTests.cs]: We should have similar check for a published standalone app as well.

File: test/dotnet-build.Tests/BuildPortableTests.cs (hunk 3 above)
Comment (livarcocc) [inline test/dotnet-build.Tests/BuildPortableTests.cs]: Just added one.

**Pull request merged**